tests/Bridge/Symfony/Validator/Metadata/Property/Restriction/PropertySchemaCollectionRestrictionTest.php

```diff
@@ -0,0 +1,121 @@
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\Collection;
+use Symfony\Component\Validator\Constraints\Email;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\NotNull;
+use Symfony\Component\Validator\Constraints\Optional;
+use Symfony\Component\Validator\Constraints\Positive;
+use Symfony\Component\Validator\Constraints\Regex;
+use Symfony\Component\Validator\Constraints\Required;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class PropertySchemaCollectionRestrictionTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private $propertySchemaCollectionRestriction;
+
+    protected function setUp(): void
+    {
+        $this->propertySchemaCollectionRestriction = new PropertySchemaCollectionRestriction([
+            new PropertySchemaLengthRestriction(),
+            new PropertySchemaRegexRestriction(),
+            new PropertySchemaFormat(),
+            new PropertySchemaCollectionRestriction(),
+        ]);
+    }
+
+    /**
+     * @dataProvider supportsProvider
+     */
+    public function testSupports(Constraint $constraint, PropertyMetadata $propertyMetadata, bool $expectedResult): void
+    {
+        self::assertSame($expectedResult, $this->propertySchemaCollectionRestriction->supports($constraint, $propertyMetadata));
+    }
+
+    public function supportsProvider(): \Generator
+    {
+        yield 'supported' => [new Collection(['fields' => []]), new PropertyMetadata(), true];
+
+        yield 'not supported' => [new Positive(), new PropertyMetadata(), false];
+    }
+
+    /**
+     * @dataProvider createProvider
+     */
+    public function testCreate(Constraint $constraint, PropertyMetadata $propertyMetadata, array $expectedResult): void
+    {
+        self::assertSame($expectedResult, $this->propertySchemaCollectionRestriction->create($constraint, $propertyMetadata));
+    }
+
+    public function createProvider(): \Generator
+    {
+        yield 'empty' => [new Collection(['fields' => []]), new PropertyMetadata(), ['type' => 'object', 'properties' => [], 'additionalProperties' => false]];
+
+        yield 'with fields' => [
+            new Collection([
+                'allowExtraFields' => true,
+                'fields' => [
+                    'name' => new Required([
+                        new NotBlank(),
+                    ]),
+                    'email' => [
+                        new NotNull(),
+                        new Length(['min' => 2, 'max' => 255]),
+                        new Email(['mode' => Email::VALIDATION_MODE_LOOSE]),
+                    ],
+                    'phone' => new Optional([
+                        new \Symfony\Component\Validator\Constraints\Type(['type' => 'string']),
+                        new Regex(['pattern' => '/^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\./0-9]*$/']),
```

Comment:
Unrelated to the feature but: you must escape the delimiter (currently Warning "preg_match(): Unknown modifier '0'").
Also no need to escape the dot inside `[]`.

```suggestion
                        new Regex(['pattern' => '/^[+]*[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s.\/0-9]*$/']),
```

Same inside `DummyCollectionValidatedEntity`

Reply:
thanks